### PR TITLE
add HTTP backend

### PR DESCRIPTION
**Pull request by sshuair**
add HTTP and HTTPs fileio support.

### CODE REVIEW

File: mmcv/fileio/file_client.py [hunk 1]
```diff
@@ -191,6 +191,20 @@ def get_text(self, filepath):
         return value_buf
 
 
+class HTTPBackend(BaseStorageBackend):
+    """HTTP and HTTPS storage bachend."""
+
+    def get(self, filepath):
+        from urllib.request import urlopen
```
Comment (hellock) [inline mmcv/fileio/file_client.py:198]: Since urllib is a builtin package, we may just put it at the front of file.

**Comment by hellock** [comment]
Unit tests are required.

File: mmcv/fileio/file_client.py (hunk 1 above)
Comment (sshuair) [reply mmcv/fileio/file_client.py:198]: Got it.

**Comment by sshuair** [comment]
@hellock I'll add some test tomorrow.

File: tests/test_fileclient.py [hunk 2]
```diff
@@ -182,6 +182,37 @@ def test_lmdb_backend(self):
...
+            http_backend.get(self.img_path)
+        with pytest.raises(Exception):
+            http_backend.get(str(self.img_path))
+        with pytest.raises(Exception):
+            http_backend.get_text(self.text_path)
+        with pytest.raises(Exception):
+            http_backend.get_text(str(self.text_path))
+
+        # input url is http image
+        http_backend.get = MagicMock(return_value=get_bytes(self.img_path))
```
Comment (hellock) [inline tests/test_fileclient.py:205]: We can use a real url to test this method, e.g., https://raw.githubusercontent.com/open-mmlab/mmcv/master/docs/mmcv-logo.png.

Comment (sshuair) [reply tests/test_fileclient.py]: It's now with real url links.

**Review by hellock — APPROVED** [review]

**Comment by hellock** [comment]
Thanks for this feature!

**Pull request merged**